packages/core/admin/_internal/node/develop.ts

```diff
@@ -24,6 +28,33 @@ interface DevelopOptions extends CLIContext {
   watchAdmin?: boolean;
 }
 
+const cleanupDistDirectory = async ({
+  tsconfig,
+  logger,
+  timer,
+}: Pick<DevelopOptions, 'tsconfig' | 'logger'> & { timer: TimeMeasurer }) => {
+  const distDir = tsconfig?.config?.options?.outDir;
+
+  if (!distDir || !(await fs.pathExists(distDir))) {
+    return;
+  }
+  const timerName = 'cleaningDist' + Date.now();
+  timer.start(timerName);
+  const cleaningSpinner = logger.spinner(`Cleaning dist dir ${distDir}`).start();
+
+  const dirContent = await fs.readdir(distDir);
+  const validFilenames = dirContent
+    // Ignore the admin build folder
+    .filter((filename) => filename !== 'build');
+  for (const filename of validFilenames) {
+    await fs.remove(path.resolve(distDir, filename));
+  }
```

Comment:
Should we try/catch this block to throw a custom error (stop the process) + fail the spinner?